zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+        up.expected_md5 = req_header.get('X-SLICE-MD5', None)
+
+        if up.slice_offset >= up.image_size:
+            raise Exception('invalid slice offset header: %s, image_size: %d' % (up.slice_offset, up.image_size))
+
+        return up
+
+    def get_upload_task(self, param):
+        # type: (UploadParam) -> UploadTask
+        task = self.upload_tasks.get_task(param.image_uuid)
+        if task is None:
+            raise Exception('image not found %s' % param.image_uuid)
```

Ruff (0.13.1)

Create your own exception

(TRY002)